### DISCUSSION TITLE
Prayer Fuel Logic

### DISCUSSION THREAD

**Discussion by DavidHarmonC4N** [DiscipleTools/disciple-tools-prayer-campaigns#192] [Q&A] [answered]
Hello,

We would like on our prayer campaign, that if there is no prayer fuel for the day, that it shows the last one posted instead of "This prayer campaign has ended".

Can someone point me in the right direction to fix this logic or help me with the change?

Thanks,
David

**Comment by corsacca** [comment]
Hey @DavidHarmonC4N 
They code for that page is here:
https://github.com/DiscipleTools/disciple-tools-prayer-campaigns/blob/master/porches/generic/site/archive-body.php

I think you are right that it is look for the post/fuel for the day, and giving the wrong message if none is found.

Maybe if none is found, and the current date is before the campaign end date, we can look look up the most recent post and display it.

**Reply by DavidHarmonC4N** [reply]
Hey @corsacca,

Thank you for replying.  I haven't really coded since 2000 and my brain almost melted looking at that logic.   It seems it could be fixed, but looking at it right now, I am not computing a change for it.  

I did however appreciate the new update that allows you to change prayer fuel from daily/weekly/etc  that goes a long way to fixing "our" issue.   I just need to get this logic fixed.  We plan to not run an end date so we want the campaign to always remain open.

I believe you are correct.  That would be ideal that if nothing is found for that "day" and the campaign end date is not reached, display the last posted.

**Comment by corsacca** [comment]
Hey @DavidHarmonC4N,

Glad the daily/weekly/etc will be helpful. Being new features, if you run into any issues let us know.

Can you give the latest form on master a try?
https://github.com/DiscipleTools/disciple-tools-prayer-campaigns/commit/3122f95e5e2ab866b3d4e59a580fba594766ffcc
That should show the latest post instead of today's post. We might look at a better solution down the road in playing with weekly prayer fuel formats.

Enjoy the changes? Consider [contributing](https://pray4movement.org/give/).

**Reply by DavidHarmonC4N** [reply]
Hey @corsacca 

I have now ran into an interesting problem.   I ran the updates yesterday and installed the new prayer plugin to all of our sites on our multisite.   It was functioning fine when I left.  Today I cannot log into any of the admin panels for the multisites.

I was going to try your update on the logic but wanted to check on something first.  If I log into my main site it works, but if I log into the backend for any subsite on the multisite it will not even let me log in.  Gives "There has been a critical error on this site."  All that has changed was the updated plugins.

![image](https://github.com/DiscipleTools/disciple-tools-prayer-campaigns/assets/140631279/73ffd948-c91c-4fc9-ae89-69251ca814d2)

**Reply by corsacca** [reply]
Hey @DavidHarmonC4N 
Sorry about that.
Some latest release have had some issue.
Are you able to access `/wp-admin/network/update-core.php` on your multisite?
Update the theme to the latest 1.43.2 and then the prayer-campaign plugin

**Reply by DavidHarmonC4N** [reply, marked as answer]
Hey @corsacca 

Joined slack.

No I cannot access that, it gives me the same error when I try to access that file.

Could I download the newest and ftp the files up into the themes dir?

David
